tests/Unit/ProtoTests/Basic/out/src/Client/BasicClient.php

```diff
@@ -144,6 +146,7 @@ public function __construct(array $options = [])
     {
         $clientOptions = $this->buildClientOptions($options);
         $this->setClientOptions($clientOptions);
+        $options = array_merge($options, $this->setEmulatorConfig($options));
```

Comment:
The main difference between `array_merge` and using `+` is that `array_merge` will merge `null` values even if the array key exists, whereas the `+` will not (if the array key exists in the first array, the second array key value is not merged in). IMHO, I think the second (using `+`) is the behavior we want, as it would allow users to override the default emulator config, e.g.:


```php
putenv('SPANNER_EMULATOR_HOST=myhost');
$spanner = new SpannerClient(['credentials' => null]); // the default credentials will be used instead of insecure ones
```

It's a very minor difference, and I can't really see anyone needing to do the above. But to me, this at least reflects what behavior I'd expect as a user.